### PR TITLE
added connected_components_using_bfs by explainig a question

### PR DESCRIPTION
**Pull request by ritik-369**
added an algorithm for finding connected components using bfs in graph section by solving a question of hackerearth
link provided in file

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: graph/connected_components_with_bfs.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            q.push(child);               // pushing non visited node in queue
+            vis[child]=1;                // marking it as visited
+         }
+      }
+   }
+   return components;                   // returning components in graph
+}
+
+
+int main()
```
Comment (Panquesito7) [inline graph/connected_components_with_bfs.cpp:69]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main()
```

File: graph/connected_components_with_bfs.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+   int connected_components;             // declaring connected_components which stores answer
+   connected_components=connected_components_bfs(adj,nodes);   // calling function
+   cout<<connected_components<<endl;                   // printing answer
+
+
+
+}
+
+
```
Comment (Panquesito7) [inline graph/connected_components_with_bfs.cpp:90]:
```suggestion
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, the documentation is not up to our [**standards**](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md). Please update the file accordingly. **Let us know if you need any help here or on our [Discord server](https://the-algorithms.com/discord/).** 😃

**Comment by ritik-369** [comment]
Thanks for Suggesting

**Comment by ritik-369** [comment]
I updated file

**Comment by ritik-369** [comment]
why it is not merged?

**Comment by ritik-369** [comment]
can i add more algorithms?

**Comment by ritik-369** [comment]
please can you help me in contribution, i am new to open source i can't understand what are these checks and why it shows merging is blocked

**Pull request closed without merging**